### PR TITLE
fix(platform)!: advanced data contract structure validation position

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
There are a few improvements that are needed to remove the ability for attacks on the network. This is one of them.

## What was done?
In this PR we removed data contract validation from check tx, and instead put it in an advanced section in block processing. If it fails we bump the identity nonce for data contract create or identity contract nonce for data contract updates.

Also bumped the min transfer amount from 1 duff to 100 duffs.

## How Has This Been Tested?
Created a test that would check_tx for an invalid contract. The check_tx passes, but the block processing marks the transaction as invalid and bumps the identity nonce as expected.

## Breaking Changes
Validation changes are breaking.

The min transfer amount is now 100 duffs.

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [X] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [X] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_credit_transfer/structure/v0/mod.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ use dpp::state_transition::identity_credit_transfer_transition::accessors::Ident
 use dpp::state_transition::identity_credit_transfer_transition::IdentityCreditTransferTransition;
 use dpp::validation::SimpleConsensusValidationResult;
 
-const MIN_TRANSFER_AMOUNT: u64 = 1000;
+const MIN_TRANSFER_AMOUNT: u64 = 100000;
```
Comment (shumkov) [inline packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_credit_transfer/structure/v0/mod.rs:11]: Why so?. Please express this change in PR description

File: packages/rs-drive-abci/tests/strategy_tests/verify_state_transitions.rs [hunk 2]
```diff
@@ -59,7 +60,7 @@ pub(crate) fn verify_state_transitions_were_or_were_not_executed(
 
             let consensus_validation_result = match state_transition.transform_into_action(
                 &platform,
-                false,
+                ValidationMode::CheckTx, //using check_tx so we don't validate state
```
Comment (shumkov) [inline packages/rs-drive-abci/tests/strategy_tests/verify_state_transitions.rs:63]: I would use more abstract name for this mode which is not related to endpoint name (layers separation).

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-drive-abci/tests/strategy_tests/verify_state_transitions.rs (hunk 2 above)
Comment (QuantumExplorer) [reply packages/rs-drive-abci/tests/strategy_tests/verify_state_transitions.rs:63]: This is only testing code, don't really want to make something just for a test.

File: packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_credit_transfer/structure/v0/mod.rs (hunk 1 above)
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_credit_transfer/structure/v0/mod.rs:11]: I mean that was 1 duff before, even 100 duffs is very low.

Comment (shumkov) [reply packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_credit_transfer/structure/v0/mod.rs:11]: Could you please describe this change in the PR description (specifically in breaking change section)

File: packages/rs-drive-abci/tests/strategy_tests/verify_state_transitions.rs (hunk 2 above)
Comment (shumkov) [reply packages/rs-drive-abci/tests/strategy_tests/verify_state_transitions.rs:63]: Ok

**Review by shumkov — APPROVED** [review]
👍
